deploy/runtools/firesim_topology_elements.py

```diff
@@ -442,68 +507,64 @@ def supernode_get_num_siblings_plus_one(self):
                 count = True
         return siblings
 
-    def supernode_get_sibling(self, siblingindex):
+    def supernode_get_sibling(self, siblingindex: int) -> FireSimDummyServerNode:
         """ return the sibling for supernode mode.
         siblingindex = 1 -> next sibling, 2 = second, 3 = last one."""
-        for index, servernode in enumerate(map( lambda x : x.get_downlink_side(), self.uplinks[0].get_uplink_side().downlinks)):
+        uplink_side = self.uplinks[0].get_uplink_side()
+        for index, servernode in enumerate(map(lambda x : x.get_downlink_side(), uplink_side.downlinks)):
             if self == servernode:
-                return self.uplinks[0].get_uplink_side().downlinks[index+siblingindex].get_downlink_side()
+                node = uplink_side.downlinks[index+siblingindex].get_downlink_side()
+                assert isinstance(node, FireSimDummyServerNode)
+                return node
+        assert False, "Should return supernode sibling"
 
-    def supernode_get_sibling_mac_address(self, siblingindex):
```

Comment:
Figured this was too verbose? I just inlined these specific calls further down the file. I can revert this if people want.